src/routes/transactions/mappers/common/safe-app-info.mapper.ts

```diff
@@ -50,8 +50,17 @@ export class SafeAppInfoMapper {
   }
 
   private getOriginUrl(transaction: MultisigTransaction): string | null {
-    return transaction.origin
-      ? JSON.parse(transaction.origin).url ?? null
-      : null;
+    if (!transaction.origin) {
+      return null;
+    }
+
+    try {
+      return JSON.parse(transaction.origin).url ?? null;
+    } catch (e) {
+      this.loggingService.debug(
+        `Safe TX Hash ${transaction.safeTxHash} origin is not valid JSON. origin=${transaction.origin}`,
+      );
+      return null;
+    }
```

Comment:
If I'm not wrong you still could use the ternary operator:
```
try {
  return transaction.origin ? JSON.parse(transaction.origin).url : null;
} catch (e) {
  this.loggingService.debug(
    `Safe TX Hash ${transaction.safeTxHash} origin is not valid JSON. origin=${transaction.origin}`,
  );
  return null;
}
```
But I'm OK with the option you find more readable 👍

Reply:
Pushed with the ternary operator [5569d8c](https://github.com/safe-global/safe-client-gateway-nest/pull/349/commits/5569d8cc0ce13374c5bd4af4c356401d9f22e379)